crates/spirv-builder/src/test/basic.rs

```diff
@@ -309,3 +309,37 @@ pub fn main(i: Input<f32>, mut o: Output<f32>) {
     o.store(i.load().signum());
 }"#);
 }
+
+#[test]
+#[ignore]
+fn allocate_const_scalar_pointer() {
+    val(r#"
+use core::ptr::Unique;
+const POINTER: Unique<[u8;4]> = Unique::<[u8; 4]>::dangling();
+
+#[allow(unused_attributes)]
+#[spirv(fragment)]
+pub fn main() {
+    let _pointer = POINTER;
+}"#);
+}
+
+#[test]
+fn allocate_vec_like_pointer() {
+    dis_fn(
+        r#"
+use core::ptr::Unique;
+const VEC_LIKE: (Unique<usize>, usize, usize) = (Unique::<usize>::dangling(), 0, 0);
+
+#[allow(unused_attributes)]
+#[spirv(fragment)]
+pub fn allocate_vec_like_pointer() {
+    let _vec_like = VEC_LIKE;
+}"#,
+        "allocate_vec_like_pointer",
+        r#"%1 = OpFunction %2 None %3
+%4 = OpLabel
+OpReturn
+OpFunctionEnd"#,
```

Comment:
There is nothing interesting going on in this function that we're disassembling here, should we use `val` instead of `dis_fn` here?

Reply:
The only reason I did this was for my own edification to understand the output better, I'll change it to val if it's not useful